forte/processors/stave/stave_processor.py

```diff
@@ -0,0 +1,316 @@
+            if response.status_code != 200:
+                return
+
+    def _default_project_configs(self):
+        # pylint: disable=line-too-long
+        """
+        Create default project configuration based on ontology.
+        This is translated from JavaScript function `createDefaultConfig` in
+        https://github.com/asyml/stave/blob/d82383de3d74bf09c0d30f33d8a902595f5aff80/src/app/pages/Projects.tsx#L140
+
+        Returns:
+            configs: A dictionary with the default config for project.
+
+        """
+        # pylint: enable=line-too-long
+
+        try:
+            ontology = self.resources.get("onto_specs_dict")
+            entry_tree = self.resources.get("merged_entry_tree")
+        except Exception as e:
+            raise ProcessorConfigError(
+                "onto_specs_dict/merged_entry_tree is"
+                " not set in resources.") from e
+
+        configs: Dict[str, Any] = {
+            "legendConfigs": {},
+            "scopeConfigs": {},
+            "layoutConfigs": {
+                "center-middle": "default-nlp",
+                "left": "default-meta",
+                "right": "default-attribute",
+                "center-bottom": "disable"
+            }
+        }
+
+        # Create legend configs
+        legend_configs: Dict[str, Any] = {}
+        entry_name_set: Set[str] = set()
+        for entry in ontology["definitions"]:
+            entry_name = entry["entry_name"]
+            entry_name_set.add(entry_name)
+            legend_configs[entry_name] = {
+                "is_selected": False,
+                "is_shown": True,
+            }
+            if "attributes" in entry and len(entry["attributes"]) > 0:
+                attributes_configs = {}
+                for attribute in entry["attributes"]:
+                    if attribute["type"] == "str":
+                        attributes_configs[attribute["name"]] = False
+                legend_configs[entry_name]["attributes"] = attributes_configs
+        configs["legendConfigs"] = legend_configs
+
+        # Find all subclass of `forte.data.ontology.top.Annotation` and
+        # update `scopeConfigs` accordingly.
+        queue = collections.deque([
+            search(entry_tree.root, "forte.data.ontology.top.Annotation")
+        ])
+        while queue:
+            size = len(queue)
+            for _ in range(size):
+                node = queue.pop()
+                if node.name in entry_name_set:
+                    configs['scopeConfigs'][node.name] = False
+                for entry in node.children:
+                    queue.appendleft(entry)
+        return configs
+
+    @classmethod
+    def default_configs(cls) -> Dict[str, Any]:
+        """
+        This defines a basic config structure for StaveProcessor.
+        :return: A dictionary with the default config for this processor.
+        Following are the keys for this dictionary:
+            - build_path: Absolute path to stave build folder.
```

Comment:
take a look at the generated sphinx doc to see if the generated docstring (especially the list) looks fine. Here is how to do this: https://github.com/asyml/forte/blob/master/.github/workflows/main.yml#L129